tripal_chado/includes/TripalFields/obi__organism/obi__organism.inc

```diff
@@ -124,19 +124,23 @@ class obi__organism extends ChadoField {
         return;
       }
       $string = $settings['field_display_string'];
+      // The infraspecific fields were introduced in Chado v1.3, include only if present.
+      // Special case conversion of the type_id 'no_rank' value to an empty string.
+      if (property_exists($organism, 'type_id')) {
+        if ($organism->type_id->name == 'no_rank') {
+          $organism->type_id->name = '';
+        }
+      }
       $label = chado_replace_tokens($string, $organism);
       $entity->{$field_name}['und'][0]['value'] = [
         $label_term => $label,
         $genus_term => $organism->genus,
         $species_term => $organism->species,
       ];
-      // The infraspecific fields were introduced in Chado v1.3.
+      // Again, only present in Chado v1.3+
       if (property_exists($organism, 'infraspecific_name')) {
-        $entity->{$field_name}['und'][0]['value'][$infraspecific_type_term] = NULL;
+        $entity->{$field_name}['und'][0]['value'][$infraspecific_type_term] = $organism->type_id->name;
         $entity->{$field_name}['und'][0]['value'][$infraspecific_name_term] = $organism->infraspecific_name;
-        if ($organism->type_id) {
-          $entity->{$field_name}['und'][0]['value'][$infraspecific_type_term] = $organism->type_id->name;
-        }
       }
```

Comment:
I think we should leave this if statement here. Not everyone will set a 'no_rank' for the type and the type_id is allowed to be null.  If it null then the `$organism->type_id->name` value will throw an error.

Reply:
Is it a problem to set these to NULL? Because infraspecific_name can be NULL too, and if it's a problem I should also check for that one.

Reply:
The if statement above (line 141) should check if the `infraspecific_name` is  NULL and short circuit the whole thing.  The internal if statement is to catch the case where type_id is NULL.   So, technically it is possible to have an infraspecific name but not indicate a type.

Reply:
The if statement above (line 141) should check if the infraspecific_name is NULL and short circuit the whole thing. The internal if statement is to catch the case where type_id is NULL. So, technically it is possible to have an infraspecific name but not indicate a type.

Reply:
Oh, but I don't think there would ever be the case that the type_id is set and the infraspecific_name is not. However, Chado allows it :-/

Reply:
Resolved by simple regex change